pom.xml

```diff
@@ -305,7 +311,38 @@
                     <version>${h2.version}</version>
                     <scope>provided</scope>
                 </dependency>
+                <dependency>
+                    <groupId>org.glassfish.jersey.core</groupId>
+                    <artifactId>jersey-common</artifactId>
+                    <version>${jersey.version}</version>
+                    <scope>provided</scope>
+                </dependency>
             </dependencies>
         </profile>
+
+        <profile>
+            <id>payara-5</id>
+            <activation>
+                <property>
+                    <name>!payara4</name>
+                </property>
+            </activation>
+            <properties>
+                <jersey.version>2.26</jersey.version>
+            </properties>
+        </profile>
+
+        <profile>
+            <id>payara-4</id>
+            <activation>
+                <property>
+                    <name>payara4</name>

```

Comment:
I don't like this.
I feel it's cleaner to just make 5 active by default, and then if you want to run 4 you specify so using the profile

Reply:
If it's active by default, specifying -Ppayara-embedded overrides it and disables both profiles. So it's between `-Ppayara-embedded,payara4` and `-Ppayara-embedded -Dpayara4`. I personally preferred the latter as it allows you to ignore it when using Payara 5, hence me doing it like this

Reply:
I far prefer just being explicit and having to do `-Ppayara-embedded,payara4` and `-Ppayara-embedded,payara5` - it's how most of our other tests work

Reply:
Can you not just set the jersey property as the 5 version as "normal" (so no 5 profile), and only override it when you specify 4?

Reply:
It now defaults to Payara 5 and will be reset to Payara 4 if the profile `payara4` is also specified.